### PR TITLE
Shade dependencies and add null checks

### PR DESCRIPTION
**Pull request by snuyanzin**
It seems we faced the situation when several connectors have similar dependencies however different versions...
As a result there is a clash of classes.

The PR makes shading of deps.
Also it was noticed that sometimes it could happen that because of error some initialization are not happened in `open` (e.g. because of clash of deps) in close there will NPE while trying to close resources

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -255,10 +291,73 @@
                                 <excludes>
                                     <exclude>org.apache.flink:force-shading</exclude>
                                     <exclude>com.google.code.findbugs:jsr305</exclude>
+                                    <exclude>com.google.code.findbugs:jsr305</exclude>
```
Comment (RyanSkraba) [inline pom.xml:294]: Duplicate line?
```suggestion
```

File: src/main/java/io/aiven/flink/connectors/bigquery/sink/BigQueryStreamingOutputFormat.java [hunk 2]
```diff
@@ -107,11 +107,17 @@ public void open(int taskNumber, int numTasks) throws IOException {
 
   @Override
   public void close() {
-    // Wait for all in-flight requests to complete.
-    inflightRequestCount.arriveAndAwaitAdvance();
+    // An error could happen before init of inflightRequestCount
```
Comment (RyanSkraba) [inline src/main/java/io/aiven/flink/connectors/bigquery/sink/BigQueryStreamingOutputFormat.java:110]: Oh nice!  This has been bugging me when running outside of the debugger!

File: pom.xml (hunk 1 above)
Comment (snuyanzin) [reply pom.xml]: yes, seems like that

**Review by RyanSkraba — APPROVED** [review]
LGTM -- the duplicate line would be better to remove, but this looks fine otherwise!  Thanks.

**Pull request merged**